specs/p2p.md

```diff
@@ -0,0 +1,17 @@
+# Peer To Peer (p2p) specifications
+## Introduction
+
+BCDBs should be able to communicate with each other to share and validate data. It means a BCDB server need to impersonate its user and be able to use his identity. A bcdb server then can create requests to other BCDB instance that are signed correctly. The receiving BCDB instance then can verify the identity
+of the caller by verifying the user signature against his public key published on the explorer. Once identity of the caller is verified, the receiver bcdb
+then can consult its local ACLs to see if the call is granted or not.
+
+While most of this logic is already in place this raises other questions:
+- How a BCDB instance find the public address of another instance. - Different ACLs that can be granted to users.
+
+## Simple hybrid p2p architecture
+- The explorer can work as a tracker, where bcdbs can retrieve public accessible addresses of other bcdb instances
+  - Explorer high availability will become an issue.
```

Comment:
A possible idea to solve this. 
Only use the explorer as bootstrap system and BCDB over time keep track of its own peer list. If you had a protocol to exchange identity information between BCDB, then you can start to have a more resilient system where BCDB can rely on each other to discover the address of new BCDB.

Reply:
Yeah, i was thinking of this but I was worried that the addresses of bcdbs might change. May be if a node is not reachable anymore or can't prove it's identity, we can drop it from the local list and consult the phonebook again.

Reply:
indeed, maintaining a peer list healthy is a task that needs to be thought out very carefully. It's not a one time thing that you can just keep forever.